docs/contributing/the_basics.md

```diff
@@ -7,65 +7,141 @@ An overview on contributing to the _Black_ project.
+The example above runs tests against Python 3.9. You can also use other versions like
+`3.7` and `pypy3`. If you don't specify a Python version, nox will look for Python
+installations for all the versions _Black_ supports and run the suite for each!
 
-### Style Changes
+By default, the test suite is ran with coverage and parallelization enabled, but you can
+disable them if they're causing you trouble:
+
+```console
+# to disable parallelization
+$ nox -s tests-3.9 -- --no-xdist
+# to disable coverage
+$ nox -s tests-3.9 -- --no-cov
+# to disable both
+$ nox -s tests-3.9 -- --no-cov --no-xdist
+```
+
+If you need to run the test suite manually, install the test dependencies in the virtual
+environment you've [previously set up](#running-black-from-source-tree) and then run
+pytest:
+
+**Linux / MacOS**
+
+```console
+(.venv)$ pip install -r tests/requirements.txt
+(.venv)$ pytest
+```
+
+**Windows**
+
+```console
+(.venv)$ python -m pip install -r tests\requirements.txt
+(.venv)$ pytest
+```
+
+### Building documentation
+
+If you make changes to docs, you can test they still build locally too.
+
+```console
+$ nox -s docs
+```
+
+If you are making many changes to docs, you may find it helpful to use the `docs-live`
+session. Each time you make a change to the docs they're rebuilt and the browser tab
+reloads.
+
+```console
+$ nox -s docs-live
+```
+
+## Style changes
 
 If a change would affect the advertised code style, please modify the documentation (The
 _Black_ code style) to reflect that change. Patches that fix unintended bugs in
 formatting don't need to be mentioned separately though. If the change is implemented
-with the `--preview` flag, please include the change in the future style document
-instead and write the changelog entry under a dedicated "Preview changes" heading.
+with the `--preview` flag, please include the change in
+{doc}`/the_black_code_style/future_style` instead and write the changelog entry under a
+dedicated "Preview changes" heading.
 
-### Docs Testing
+## Changelog requirement
 
-If you make changes to docs, you can test they still build locally too.
+_Black_ has CI that will check for an entry corresponding to your PR in `CHANGES.md`. If
+you feel this PR does not require a changelog entry please state that in a comment and a
+maintainer can add a `skip news` label to make the CI pass. Otherwise, please ensure you
+have a line in the following format:
 
-```console
-(.venv)$ pip install -r docs/requirements.txt
-(.venv)$ pip install [-e] .[d]
-(.venv)$ sphinx-build -a -b html -W docs/ docs/_build/
+```md
+- `Black` is now more awesome (#X)
 ```
 
+Note that X should be your PR number, not issue number! To workout X, please use
+[Next PR Number](https://ichard26.github.io/next-pr-number/?owner=psf&name=black). This
+is not perfect but saves a lot of release overhead as now the releaser does not need to
+go back and workout what to add to the `CHANGES.md` for each release.
```

Comment:
```suggestion
Note that X should be your PR number, not issue number! To work out X, please use
[Next PR Number](https://ichard26.github.io/next-pr-number/?owner=psf&name=black). This
saves a lot of release overhead as now the releaser does not need to
go back and work out what to add to the `CHANGES.md` for each release.
```
No need to apologize here

Reply:
That said, wouldn't it be worth implementing this as a pre-commit script or GHA ? (not necessarily in this PR) "If the diff in CHANGES.md is a single added line, then add `(#PR)` at the end."
(depending on if you decide on a different path on #2766)